pkg/api/converterfromapi.go

```diff
@@ -182,10 +190,18 @@ func convertPropertiesToV20160330(api *Properties, p *v20160330.Properties) {
 		convertMasterProfileToV20160330(api.MasterProfile, p.MasterProfile)
 	}
 	p.AgentPoolProfiles = []*v20160330.AgentPoolProfile{}
-	for _, apiProfile := range api.AgentPoolProfiles {
-		v20160330Profile := &v20160330.AgentPoolProfile{}
-		convertAgentPoolProfileToV20160330(apiProfile, v20160330Profile)
-		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20160330Profile)
+	// We expect exactly 2 agent pools: index [0] is the private pool, index [1] is the public pool
+	for i, apiProfile := range api.AgentPoolProfiles {
+		// a V20160330 should only have one (private) agent pool
+		if i < 1 {
+			v20160330Profile := &v20160330.AgentPoolProfile{}
```

Comment:
Is there an identifying property inside the agent pool itself that we can use to determine this (i.e., `"type": "private"`)? I'm not sure there is. (unless we use this PR to introduce such an identifying property)